pilot/pkg/bootstrap/server.go

```diff
@@ -589,10 +534,19 @@ func (s *Server) initDNSTLSListener(dns string, tlsOptions TLSOptions) error {
 	return nil
 }
 
-// initialize secureGRPCServer - using DNS certs
-func (s *Server) initSecureGrpcServer(port string, keepalive *istiokeepalive.Options, tlsOptions TLSOptions) error {
+// initialize secureGRPCServer.
+func (s *Server) initSecureGrpcServer(args *PilotArgs, port string) error {
+	if features.IstiodService.Get() == "" {
```

Comment:
I think we are passed the point the flag can be off - it was used in early 1.5 to turn off Istiod and only use old pilot behavior, but now old pilot is no longer used. 

I know you're just moving code around and refactoring.